src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs

```diff
@@ -51,73 +51,89 @@ public FullNodeFeatureExecutor(IFullNode fullNode, ILoggerFactory loggerFactory)
         /// <inheritdoc />
         public void Initialize()
         {
+            this.logger.LogTrace("()");
+
             try
             {
                 this.Execute(service => service.ValidateDependencies(this.node.Services));
                 this.Execute(service => service.Initialize());
             }
-            catch (Exception ex)
+            catch (AggregateException)
             {
-                this.logger.LogError("An error occurred starting the application: {0}", ex);
+                this.logger.LogError("An error occurred starting the application.");
+                this.logger.LogTrace("(-)[INITIALIZE_EXCEPTION]");
                 throw;
             }
+
+            this.logger.LogTrace("(-)");
         }
 
         /// <inheritdoc />
         public void Dispose()
         {
+            this.logger.LogTrace("()");
+
             try
             {
                 this.Execute(feature => feature.Dispose(), true);
             }
-            catch (Exception ex)
+            catch (AggregateException)
```

Comment:
I don't think you have to specify `AggregateException` here.

Reply:
Agree

Reply:
yes also at the other method otherwise looks good

Reply:
Fixed